egs2/TEMPLATE/tts1/tts.sh

```diff
@@ -620,18 +620,40 @@ if [ ${stage} -le 7 ] && [ ${stop_stage} -ge 7 ]; then
     _feats_type="$(<${data_feats}/${train_set}/feats_type)"
 
     # NOTE(kamo): If feats_type=raw, vocoder_conf is unnecessary
-    if [ "${_feats_type}" == fbank ] || [ "${_feats_type}" == stft ]; then
+    _scp=wav.scp
+    _type=sound
+    if [ "${_feats_type}" = fbank ] || [ "${_feats_type}" = stft ]; then
         _opts+="--vocoder_conf n_fft=${n_fft} "
         _opts+="--vocoder_conf n_shift=${n_shift} "
         _opts+="--vocoder_conf win_length=${win_length} "
         _opts+="--vocoder_conf fs=${fs} "
+        _scp=feats.scp
+        _type=kaldi_ark
     fi
-    if [ "${_feats_type}" == fbank ]; then
+    if [ "${_feats_type}" = fbank ]; then
         _opts+="--vocoder_conf n_mels=${n_mels} "
         _opts+="--vocoder_conf fmin=${fmin} "
         _opts+="--vocoder_conf fmax=${fmax} "
     fi
 
+    # If we use teacher forcing or GST embedding in inference,
+    # it is necessary to pass both speech and text
+    # TODO(kan-bayashi): Need to check train_args and decode_args
+    _use_speech=false
+    _use_teacher_forcing=false
+    _use_gst=false
+    if [ -n "${train_config}" ]; then
+        _use_gst=$(pyscripts/utils/get_yaml.py \
+            "${train_config}" tts_conf.use_gst)
+    fi
+    if [ -n "${decode_config}" ]; then
+        _use_teacher_forcing=$(pyscripts/utils/get_yaml.py \
+            "${decode_config}" use_teacher_forcing)
+    fi
+    if [ "${_use_teacher_forcing,,}" = "true" ] || [ "${_use_gst,,}" = "true" ]; then
+        _use_speech=true
+    fi
```

Comment:
My suggestion:

1. Always give `--data_path_and_name_and_type ${_data}/${_scp},speech,${_type}`
2. Add something like `tts.use_speech` as an attribute or public method to the interface to check `use_speech` option
3. Then

    ```python
    if not tts.use_speech:
        data_path_and_name_type = list(filter(lambda x: x[1] != "speech", data_path_and_name_type))
```

Reply:
OK. I will follow your suggestions!